### PR TITLE
Implement rr-run-c

### PR DESCRIPTION
**Pull request by alehander92**
Fix https://github.com/mozilla/rr/issues/2399 
it seems i can do it without a python command but maybe cleaner that way?
i moved python macros in front, no idea if its a problem

### CODE REVIEW

**Comment by alehander92** [comment]
sorry i didnt do it that way (no python and no tests(i have to run some locally i guess)), ill wait for a review

**Comment by rocallahan** [comment]
This resets the language to `auto`. That's going to break people who use `set lang ...` right? That seems bad.

**Comment by alehander92** [comment]
you're right, i thought i can simplify it that way, i'll use show language then

**Comment by alehander92** [comment]
i hope parsing it is fine

**Comment by alehander92** [comment]
ok, force pushed, thank you, i also generalize to rr-run-c because of run hook

File: src/GdbCommandHandler.cc [hunk 1]
```diff
@@ -41,6 +41,8 @@ python
 
 import re
 
+
+
```
Comment (rocallahan) [inline src/GdbCommandHandler.cc]: Drop this change

File: src/GdbCommandHandler.cc [hunk 2]
```diff
@@ -105,6 +107,31 @@ class RRCmd(gdb.Command):
...
+    start = output.find('"')
+    end = output.find('"', start + 1)
+    user_lang = output[start + 1 : end]
+    if user_lang.startswith('auto;'): # auto currently name
+      user_lang = user_lang[user_lang.rfind(' ') : ]
+    if arg and arg[0] == '"' and arg[-1] == '"':
+      arg = arg[1:-1].replace('\\n', '\n')
+    gdb.execute('set lang c')
+    gdb.execute(arg)
+    gdb.execute('set lang %s' % user_lang)
```
Comment (rocallahan) [inline src/GdbCommandHandler.cc]: If the current language is `auto` then this sets it to whatever the auto language is, e.g. `c`. Instead, we should do `set lang auto` again.

Comment (alehander92) [reply src/GdbCommandHandler.cc]: yeah !

**Comment by rocallahan** [comment]
This looks good. Can you squash the commits into one? Then I'll merge it.

**Comment by alehander92** [comment]
Thank you ! I pushed

**Comment by rocallahan** [comment]
Thanks!

**Pull request merged**